platforms/solana/src/platform.ts

```diff
@@ -176,41 +178,66 @@ export class SolanaPlatform<N extends Network>
     stxns: SignedTx[],
     opts?: SendOptions,
   ): Promise<TxHash[]> {
-    const { blockhash, lastValidBlockHeight } = await this.latestBlock(rpc);
-    const txhashes = await Promise.all(
-      stxns.map((stxn) =>
-        rpc.sendRawTransaction(
-          stxn,
-          // Set the commitment level to match the rpc commitment level
-          // otherwise, it defaults to finalized
-          opts ?? { preflightCommitment: rpc.commitment },
-        ),
-      ),
-    );
-
     const results = await Promise.all(
-      txhashes.map((signature) => {
-        return rpc.confirmTransaction(
-          {
-            signature,
-            blockhash,
-            lastValidBlockHeight,
-          },
-          rpc.commitment,
-        );
-      }),
+      stxns.map((stxn) => this.sendTxWithRetry(rpc, stxn)),
     );
 
+    const txhashes = results.map((r) => r.signature);
+
     const erroredTxs = results
-      .filter((result) => result.value.err)
-      .map((result) => result.value.err);
+      .filter((r) => r.response.value.err)
+      .map((r) => r.response.value.err);
 
     if (erroredTxs.length > 0)
       throw new Error(`Failed to confirm transaction: ${erroredTxs}`);
 
     return txhashes;
   }
 
+  static async sendTxWithRetry(
+    rpc: Connection,
+    tx: SignedTx,
+    retryInterval = 5000,
+  ): Promise<{
+    signature: string;
+    response: RpcResponseAndContext<SignatureResult>;
+  }> {
+    const sendOptions: SendOptions = {
+      skipPreflight: true, // TODO: need to make sure tx was simulated just before sending
```

Comment:
We are skipping preflight here (important), but this is different than the previous behavior. Previously the call to `sendRawTransaction` would set `skipPreflight: false` by default. So now we either need to simulate the transaction here just before sending it or expect the caller to have done it. Callers already simulate the tx to get the compute budget, so simulating it again here would be redundant. Maybe we can add a flag to `determineComputeBudget` to throw when simulation fails. Not sure what's best here

Reply:
Actually, I think we could set `skipPreflight: false` for the first call to `sendRawTransaction` so it throws if simulation fails and then set it to `true` on any subsequent calls.